src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php

```diff
@@ -93,6 +93,12 @@ private function safelyUnserialize(string $contents): Envelope
         try {
             /** @var Envelope */
             $envelope = unserialize($contents);
+        } catch (\Throwable $e) {
+            if ($e instanceof MessageDecodingFailedException) {
+                throw $e;
+            }
+
+            throw new MessageDecodingFailedException(sprintf('Could not decode message using PHP serialization: "%s".', $contents), previous: $e);
```

Comment:
I wouldn't put the contents in the error message, it can contain sensitive data.

```suggestion
            throw new MessageDecodingFailedException('Could not unserialize Envelope: '.$e->getMessage(), $contents), previous: $e);
```

Reply:
Agree :+1: 

FYI the content is attached to the `$signalException`.

Reply:
Maybe We should remove it from it as well?

Reply:
I removed it while merging.